File: eAnimalShelter/UI/eanimalshelter_desktop/lib/models/requests/animal_update_request.dart
import 'animal_insert_request.dart';

class AnimalUpdateRequest
    extends AnimalInsertRequest {
  AnimalUpdateRequest({
    required super.name,
    required super.speciesId,
    required super.breedId,
    required super.gender,
    required super.birthDate,
    required super.description,
    required super.personalityDescription,
    required super.healthStatus,
    required super.isVaccinated,
    super.medicalNotes,
    super.arrivalDate,
    required super.adoptionStatus,
  });
}

File: eAnimalShelter/eAnimalShelter.Model/Requests/AnimalUpdateRequest.cs
using eAnimalShelter.Model.Enums;

namespace eAnimalShelter.Model.Requests
{
    public class AnimalUpdateRequest
    {
        public string Name { get; set; } = string.Empty;

        public int SpeciesId { get; set; }

        public int BreedId { get; set; }

        public AnimalGender Gender { get; set; } 

        public DateTime BirthDate { get; set; }

        public string Description { get; set; } = string.Empty;

        public string PersonalityDescription { get; set; } = string.Empty;

        public string HealthStatus { get; set; } = string.Empty;

        public bool IsVaccinated { get; set; }

        public string? MedicalNotes { get; set; }

        public DateTime? ArrivalDate { get; set; }

        public AnimalStatus AdoptionStatus { get; set; }
    }
}